File: ios/Releaf/DesignSystem/Components/PagePreviewRow.swift
/*
 * PagePreviewRow.swift
 *
 * Compact row for a single page preview. Appears inside chapter sections
 * in notebook detail and on the home screen's "Continue" list.
 *
 * Layout:
 *   [icon chip]  Title                     [thumbnail
 *                Description                with photo
 *                Updated · 2 hours ago       count badge]
 *
 * Pure presentation — no tap handling. Wrap the row in a
 * `NavigationLink(value:)` or a `Button` at the call site to make it
 * tappable. Inside a NavigationLink, set `.buttonStyle(.plain)` on the
 * link so the row's own visuals show through.
 *
 * Ported from Inkcreate mobile DS.
 */

import SwiftUI

public struct PagePreviewRow: View {
    public let title: String
    public let description: String?
    public let meta: String
    public let systemIcon: String
    public let thumbnailImage: Image?
    public let photoCount: Int

    public init(
        title: String,
        description: String? = nil,
        meta: String,
        systemIcon: String = "doc.text",
        thumbnailImage: Image? = nil,
        photoCount: Int = 0
    ) {
        self.title = title
        self.description = description
        self.meta = meta
        self.systemIcon = systemIcon
        self.thumbnailImage = thumbnailImage
        self.photoCount = photoCount
    }

    public var body: some View {
        HStack(alignment: .top, spacing: AppSpacing.s3) {
            iconChip

            VStack(alignment: .leading, spacing: 2) {
                Text(title)
                    .font(AppText.button)
                    .foregroundColor(AppColors.textPrimary)
                    .lineLimit(1)
                if let description {
                    Text(description)
                        .font(AppText.meta)
                        .foregroundColor(AppColors.textSecondary)
                        .lineLimit(2)
                }
                Text(meta)
                    .font(AppText.meta)
                    .foregroundColor(AppColors.textTertiary)
                    .lineLimit(1)
            }
            .frame(maxWidth: .infinity, alignment: .leading)

            thumbnail
        }
        .padding(AppSpacing.s3)
        .background(
            RoundedRectangle(cornerRadius: AppRadius.md, style: .continuous)
                .fill(AppColors.cardSolid)
        )
        .overlay(
            RoundedRectangle(cornerRadius: AppRadius.md, style: .continuous)
                .stroke(AppColors.borderDefault, lineWidth: 1)
        )
        .appShadow(.xs)
        .contentShape(Rectangle())
    }

    // MARK: Pieces

    private var iconChip: some View {
        ZStack {
            RoundedRectangle(cornerRadius: AppRadius.sm, style: .continuous)
                .fill(AppColors.coralSoft)
            Image(systemName: systemIcon)
                .font(.system(size: 16, weight: .semibold))
                .foregroundColor(AppColors.coralDeep)
        }
        .frame(width: 36, height: 36)
    }

    @ViewBuilder
    private var thumbnail: some View {
        if thumbnailImage != nil || photoCount > 0 {
            ZStack(alignment: .bottomTrailing) {
                RoundedRectangle(cornerRadius: AppRadius.sm, style: .continuous)
                    .fill(AppColors.subtle)
                if let thumbnailImage {
                    thumbnailImage
                        .resizable()
                        .aspectRatio(contentMode: .fill)
                        .clipShape(RoundedRectangle(cornerRadius: AppRadius.sm, style: .continuous))
                }
                if photoCount > 1 {
                    Text("+\(photoCount - 1)")
                        .font(AppText.tag)
                        .foregroundColor(AppColors.textOnAccent)
                        .padding(.horizontal, 6)
                        .padding(.vertical, 2)
                        .background(Capsule().fill(AppColors.textPrimary.opacity(0.72)))
                        .padding(4)
                }
            }
            .frame(width: 52, height: 52)
        }
    }
}

#if DEBUG
struct PagePreviewRow_Previews: PreviewProvider {
    static var previews: some View {
        VStack(spacing: AppSpacing.s2) {
            PagePreviewRow(
                title: "Sunday, Apr 19 — Page 1",
                description: "Standup notes, follow-ups on the auth rework.",
                meta: "Updated 2 hours ago",
                photoCount: 3
            )
            PagePreviewRow(
                title: "Saturday grocery run",
                meta: "Updated yesterday"
            )
        }
        .padding(AppSpacing.s4)
        .background(AppColors.canvas)
        .frame(width: 390)
    }
}
#endif

File: android/app/src/main/java/app/releaf/mobile/ui/components/PagePreviewRow.kt
/*
 * PagePreviewRow.kt
 *
 * Compact row for a page preview inside chapter sections or the home
 * "Continue" list.
 *
 * Ported from Inkcreate mobile DS.
 */

package app.releaf.mobile.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.Description
import androidx.compose.material3.Icon
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextOverflow
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import app.releaf.mobile.ui.theme.AppColors
import app.releaf.mobile.ui.theme.AppAccent
import app.releaf.mobile.ui.theme.AppRadius
import app.releaf.mobile.ui.theme.AppSpacing
import app.releaf.mobile.ui.theme.AppTypography

@Composable
fun PagePreviewRow(
    title: String,
    meta: String,
    modifier: Modifier = Modifier,
    description: String? = null,
    icon: ImageVector = Icons.Filled.Description,
    photoCount: Int = 0,
    onClick: () -> Unit = {},
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(AppRadius.md))
            .background(AppColors.CardSolid)
            .border(1.dp, AppColors.BorderDefault, RoundedCornerShape(AppRadius.md))
            .clickable(onClick = onClick)
            .padding(AppSpacing.s3),
        horizontalArrangement = Arrangement.spacedBy(AppSpacing.s3),
        verticalAlignment = Alignment.Top,
    ) {
        IconChip(icon = icon)

        Column(
            modifier = Modifier.weight(1f),
            verticalArrangement = Arrangement.spacedBy(2.dp),
        ) {
            Text(
                text = title,
                style = AppTypography.Button,
                color = AppColors.TextPrimary,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
            if (description != null) {
                Text(
                    text = description,
                    style = AppTypography.Meta,
                    color = AppColors.TextSecondary,
                    maxLines = 2,
                    overflow = TextOverflow.Ellipsis,
                )
            }
            Text(
                text = meta,
                style = AppTypography.Meta,
                color = AppColors.TextTertiary,
                maxLines = 1,
                overflow = TextOverflow.Ellipsis,
            )
        }

        if (photoCount > 0) {
            Thumbnail(photoCount = photoCount)
        }
    }
}

@Composable
private fun IconChip(icon: ImageVector) {
    Box(
        modifier = Modifier
            .size(36.dp)
            .clip(RoundedCornerShape(AppRadius.sm))
            .background(AppAccent.soft),
        contentAlignment = Alignment.Center,
    ) {
        Icon(
            imageVector = icon,
            contentDescription = null,
            tint = AppAccent.deep,
            modifier = Modifier.size(16.dp),
        )
    }
}

@Composable
private fun Thumbnail(photoCount: Int) {
    Box(
        modifier = Modifier
            .size(52.dp)
            .clip(RoundedCornerShape(AppRadius.sm))
            .background(AppColors.Subtle),
        contentAlignment = Alignment.BottomEnd,
    ) {
        if (photoCount > 1) {
            Box(
                modifier = Modifier
                    .padding(4.dp)
                    .clip(RoundedCornerShape(AppRadius.pill))
                    .background(AppColors.TextPrimary.copy(alpha = 0.72f))
                    .padding(horizontal = 6.dp, vertical = 2.dp),
            ) {
                Text(
                    "+${photoCount - 1}",
                    style = AppTypography.Tag,
                    color = AppColors.TextOnAccent,
                )
            }
        }
    }
}

@Preview(showBackground = true, backgroundColor = 0xFFF5EEE3, widthDp = 390)
@Composable
private fun PagePreviewRowPreview() {
    Column(
        modifier = Modifier
            .background(AppColors.Canvas)
            .padding(AppSpacing.s4),
        verticalArrangement = Arrangement.spacedBy(AppSpacing.s2),
    ) {
        PagePreviewRow(
            title = "Sunday, Apr 19 — Page 1",
            description = "Standup notes, follow-ups on the auth rework.",
            meta = "Updated 2 hours ago",
            photoCount = 3,
        )
        PagePreviewRow(
            title = "Saturday grocery run",
            meta = "Updated yesterday",
        )
    }
}
